src/swupd/hash_test.go

```diff
@@ -27,3 +30,22 @@ func TestInternHash(t *testing.T) {
 		})
 	}
 }
+
+func TestHashPrinting(t *testing.T) {
+	s := "0000000000000000000000000000000000000000000000000000000000000001"
+	v := internHash(s)
+	sout := fmt.Sprintf("%v", v)
+	if sout != s {
+		t.Errorf("in and out of hashtable do not match\n\t%v\n\t%v", sout, s)
+	}
+}
+
+func TestHashPrinting2(t *testing.T) {
+	s := []byte("0000000000000000000000000000000000000000000000000000000000000001")
+	v := internHash(string(s))
+	s[0] = '1'
+	sout := fmt.Sprintf("%v", v)
+	if sout == string(s) {
+		t.Errorf("in and out of hashtable do not match\n\t%v\n\t%v", sout, s)
```

Comment:
Error message should say "do match" here I think.

Reply:
Yes.
Actually this test is pretty bogus.